RecoTracker/MkFitCore/interface/Hit.h

```diff
@@ -0,0 +1,279 @@
+    static void reset();
+  };
+  typedef std::vector<MCHitInfo> MCHitInfoVec;
+
+  struct MeasurementState {
+  public:
+    MeasurementState() {}
+    MeasurementState(const SVector3& p, const SVector6& e) : pos_(p), err_(e) {}
+    MeasurementState(const SVector3& p, const SMatrixSym33& e) : pos_(p) {
+      for (int i = 0; i < 6; ++i)
+        err_[i] = e.Array()[i];
+    }
+    const SVector3& parameters() const { return pos_; }
+    SMatrixSym33 errors() const {
+      SMatrixSym33 result;
+      for (int i = 0; i < 6; ++i)
+        result.Array()[i] = err_[i];
+      return result;
+    }
+    SVector3 pos_;
+    SVector6 err_;
+  };
+
+  class Hit {
+  public:
+    Hit() : mcHitID_(-1) {}
+
+    Hit(const SVector3& position, const SMatrixSym33& error, int mcHitID = -1)
+        : state_(position, error), mcHitID_(mcHitID) {}
+
+    const SVector3& position() const { return state_.parameters(); }
+    const SVector3& parameters() const { return state_.parameters(); }
+    const SMatrixSym33 error() const { return state_.errors(); }
+
+    const float* posArray() const { return state_.pos_.Array(); }
+    const float* errArray() const { return state_.err_.Array(); }
+
+    // Non-const versions needed for CopyOut of Matriplex.
+    SVector3& parameters_nc() { return state_.pos_; }
+    SVector6& error_nc() { return state_.err_; }
+
+    float r() const {
+      return sqrtf(state_.parameters().At(0) * state_.parameters().At(0) +
+                   state_.parameters().At(1) * state_.parameters().At(1));
+    }
+    float x() const { return state_.parameters().At(0); }
+    float y() const { return state_.parameters().At(1); }
+    float z() const { return state_.parameters().At(2); }
+    float exx() const { return state_.errors().At(0, 0); }
+    float eyy() const { return state_.errors().At(1, 1); }
+    float ezz() const { return state_.errors().At(2, 2); }
+    float phi() const { return getPhi(state_.parameters().At(0), state_.parameters().At(1)); }
+    float eta() const {
+      return getEta(state_.parameters().At(0), state_.parameters().At(1), state_.parameters().At(2));
+    }
+    float ephi() const { return getPhiErr2(x(), y(), exx(), eyy(), state_.errors().At(0, 1)); }
+    float eeta() const {
+      return getEtaErr2(x(),
+                        y(),
+                        z(),
+                        exx(),
+                        eyy(),
+                        ezz(),
+                        state_.errors().At(0, 1),
+                        state_.errors().At(0, 2),
+                        state_.errors().At(1, 2));
+    }
+
+    const MeasurementState& measurementState() const { return state_; }
+
+    int mcHitID() const { return mcHitID_; }
+    int layer(const MCHitInfoVec& globalMCHitInfo) const { return globalMCHitInfo[mcHitID_].layer(); }
+    int mcTrackID(const MCHitInfoVec& globalMCHitInfo) const { return globalMCHitInfo[mcHitID_].mcTrackID(); }
+
+    static constexpr int kMinChargePerCM = 1620;
```

Comment:
good point, I've added it to the list of things to make configurable.
This specific constant is likely to stay though because it's also used to truncate the local compressed hit charge/cm value (8 bits; to keep memory use low).